File: app/src/main/java/com/gl/revival/ProfilePanel.java
package com.gl.revival;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.ScrollView;
import android.widget.TextView;
import android.widget.Button;
import android.widget.Toast;
import java.util.List;

public class ProfilePanel extends Activity {
    private LinearLayout listContainer;
    private ProfileManager pm;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        pm = new ProfileManager(this);
        buildUi();
        refreshList();
    }

    private void buildUi() {
        ScrollView sv = new ScrollView(this);
        LinearLayout root = new LinearLayout(this);
        root.setOrientation(LinearLayout.VERTICAL);
        root.setPadding(32, 32, 32, 32);
        root.setBackgroundColor(iOSColorPalette.systemBackground(this));

        TextView title = new TextView(this);
        title.setText("Profile Manager");
        title.setTextSize(22f);
        title.setTextColor(iOSColorPalette.label(this));
        root.addView(title);

        Button btnNew = new Button(this);
        btnNew.setText("+ New Profile");
        btnNew.setBackgroundColor(iOSColorPalette.systemGreen(this));
        btnNew.setTextColor(0xFFFFFFFF);
        btnNew.setOnClickListener(v -> {
            Profile p = pm.createDefault();
            pm.setActive(p.id);
            refreshList();
            Toast.makeText(this, "Created " + p.name, Toast.LENGTH_SHORT).show();
        });
        root.addView(btnNew);

        Button btnImport = new Button(this);
        btnImport.setText("Import / Export JSON");
        btnImport.setBackgroundColor(iOSColorPalette.systemIndigo(this));
        btnImport.setTextColor(0xFFFFFFFF);
        btnImport.setOnClickListener(v -> {
            startActivity(new Intent(this, ConfigActivity.class));
        });
        root.addView(btnImport);

        listContainer = new LinearLayout(this);
        listContainer.setOrientation(LinearLayout.VERTICAL);
        listContainer.setPadding(0, 24, 0, 0);
        root.addView(listContainer);

        sv.addView(root);
        setContentView(sv);
    }

    private void refreshList() {
        listContainer.removeAllViews();
        List<Profile> list = pm.listProfiles();
        String activeId = pm.getActiveProfileId();
        for (Profile p : list) {
            listContainer.addView(makeCard(p, p.id.equals(activeId)));
        }
    }

    private ViewGroup makeCard(Profile p, boolean active) {
        LinearLayout card = new LinearLayout(this);
        card.setOrientation(LinearLayout.VERTICAL);
        card.setPadding(24, 24, 24, 24);
        card.setBackgroundColor(active
            ? iOSColorPalette.systemBlueLow(this)
            : iOSColorPalette.secondarySystemBackground(this));
        LinearLayout.LayoutParams lp = new LinearLayout.LayoutParams(-1, -2);
        lp.setMargins(0, 0, 0, 16);
        card.setLayoutParams(lp);

        TextView name = new TextView(this);
        name.setText((active ? "● " : "○ ") + p.name);
        name.setTextSize(16f);
        name.setTextColor(iOSColorPalette.label(this));
        card.addView(name);

        TextView meta = new TextView(this);
        meta.setText(p.gpuRenderer + " / " + p.ramMb + "MB");
        meta.setTextSize(12f);
        meta.setTextColor(iOSColorPalette.secondaryLabel(this));
        card.addView(meta);

        LinearLayout row = new LinearLayout(this);
        row.setOrientation(LinearLayout.HORIZONTAL);

        Button btnUse = new Button(this);
        btnUse.setText("Activate");
        btnUse.setOnClickListener(v -> {
            pm.setActive(p.id);
            refreshList();
            GLLog.i("ProfilePanel", "Activated " + p.id);
        });
        row.addView(btnUse);

        Button btnDel = new Button(this);
        btnDel.setText("Delete");
        btnDel.setOnClickListener(v -> {
            pm.deleteProfile(p.id);
            refreshList();
            GLLog.i("ProfilePanel", "Deleted " + p.id);
        });
        row.addView(btnDel);

        card.addView(row);
        return card;
    }
}

File: src/main/java/com/gl/revival/ProfilePanel.kt
package com.gl.revival

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.ViewGroup
import android.widget.Button
import android.widget.LinearLayout
import android.widget.ScrollView
import android.widget.TextView
import android.widget.Toast

class ProfilePanel : Activity() {
    private lateinit var listContainer: LinearLayout
    private val pm by lazy { ProfileManager(this) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        buildUi()
        refreshList()
    }

    private fun buildUi() {
        val root = LinearLayout(this).apply {
            orientation = LinearLayout.VERTICAL
            setPadding(32, 32, 32, 32)
            setBackgroundColor(iOSColorPalette.systemBackground(this@ProfilePanel))
        }

        root.addView(TextView(this).apply {
            text = "Profile Manager"
            textSize = 22f
            setTextColor(iOSColorPalette.label(this@ProfilePanel))
        })

        root.addView(Button(this).apply {
            text = "+ New Profile"
            setBackgroundColor(iOSColorPalette.systemGreen(this@ProfilePanel))
            setTextColor(0xFFFFFFFF.toInt())
            setOnClickListener {
                val p = pm.createDefault()
                pm.setActive(p.id)
                refreshList()
                Toast.makeText(this@ProfilePanel, "Created ${p.name}", Toast.LENGTH_SHORT).show()
            }
        })

        root.addView(Button(this).apply {
            text = "Import / Export JSON"
            setBackgroundColor(iOSColorPalette.systemIndigo(this@ProfilePanel))
            setTextColor(0xFFFFFFFF.toInt())
            setOnClickListener {
                startActivity(Intent(this@ProfilePanel, ConfigActivity::class.java))
            }
        })

        listContainer = LinearLayout(this).apply {
            orientation = LinearLayout.VERTICAL
            setPadding(0, 24, 0, 0)
        }
        root.addView(listContainer)

        setContentView(ScrollView(this).apply { addView(root) })
    }

    private fun refreshList() {
        listContainer.removeAllViews()
        val activeId = pm.activeProfileId
        pm.listProfiles().forEach { p ->
            listContainer.addView(makeCard(p, p.id == activeId))
        }
    }

    private fun makeCard(p: Profile, active: Boolean): ViewGroup {
        val card = LinearLayout(this).apply {
            orientation = LinearLayout.VERTICAL
            setPadding(24, 24, 24, 24)
            setBackgroundColor(
                if (active) iOSColorPalette.systemBlueLow(this@ProfilePanel)
                else iOSColorPalette.secondarySystemBackground(this@ProfilePanel)
            )
            layoutParams = LinearLayout.LayoutParams(-1, -2).apply {
                setMargins(0, 0, 0, 16)
            }
        }

        card.addView(TextView(this).apply {
            text = "${if (active) "● " else "○ "}${p.name}"
            textSize = 16f
            setTextColor(iOSColorPalette.label(this@ProfilePanel))
        })

        card.addView(TextView(this).apply {
            text = "${p.gpuRenderer} / ${p.ramMb}MB"
            textSize = 12f
            setTextColor(iOSColorPalette.secondaryLabel(this@ProfilePanel))
        })

        val row = LinearLayout(this).apply { orientation = LinearLayout.HORIZONTAL }

        row.addView(Button(this).apply {
            text = "Activate"
            setOnClickListener {
                pm.setActive(p.id)
                refreshList()
                GLLog.i("ProfilePanel", "Activated ${p.id}")
            }
        })

        row.addView(Button(this).apply {
            text = "Delete"
            setOnClickListener {
                pm.deleteProfile(p.id)
                refreshList()
                GLLog.i("ProfilePanel", "Deleted ${p.id}")
            }
        })

        card.addView(row)
        return card
    }
}
